src/utils/get-search.js

```diff
@@ -0,0 +1,12 @@
+export function getSearch(parsedURL) {
```

Comment:
I'm not sure a helper function is needed for this. It can be written concisely without any function calls or allocations like so:
```js
u.search || u.href[u.href.length - u.hash.length - 1] === '?' ? '?' : ''
```

Reply:
I can do that, but I usually avoid "magic looking code" (or just unreadable at first sight) even if I personally like that. Decision is up to node-fetch authors.

Reply:
Oh, and your code will fail, it should look for `?` with `includes`

Reply:
Hmm... What input will cause it to fail? Using `includes` will perform a linear search from the beginning of the string. Given that we know the input must have been valid to have been parsed by URL, we should be able to safely check the end of the `href`, minus the `hash`.

Reply:
It will fail for http://example.com/?# but if you include some hash check too it should be okay to test only the last character I think

Reply:
Ah, good spotting 😄 I suppose unit tests would be useful for covering such edge cases. How about this?

```js
function getSearch(u) {
  if (u.search) {
    return u.search;
  }
  const lastOffset = u.href.length - 1;
  const hash = u.hash || u.href[lastOffset] === '#' ? '#' : '';
  return u.href[lastOffset - hash.length] === '?' ? '?' : '';
}
```

Reply:
Seems good to me. I have added an unit test for the search string already, so updating my solution with yours should pass the tests

I don't know if just the hash alone causes issues anywhere, as it is not send to the server anyway

Reply:
I have pushed an update with your suggestion (and fixed missing parentheses). I have replaced `u` with `parsedURL` for readability (build job should minify the code, seems like builds are kind of work in progress so I guess this will come later).

Thank you for your suggestions